Tasks/Gradle/CodeAnalysis/Common/BaseTool.ts

```diff
@@ -0,0 +1,153 @@
+import {AnalysisResult} from './AnalysisResult'
+import {IAnalysisTool} from './IAnalysisTool'
+import {BuildOutput, BuildEngine} from './BuildOutput'
+import {ModuleOutput} from './ModuleOutput'
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
+
+import path = require('path');
+import fs = require('fs');
+import glob = require('glob');
+import xml2js = require('xml2js');
+
+import tl = require('vsts-task-lib/task');
+
+
+/**
+ * An abstract class that is the base for both configuring a build to use an analysis tool and   
+ * 
+ * @export
+ * @class PmdReportParser
+ * @implements {IAnalysisToolReportParser}
+ */
+export abstract class BaseTool implements IAnalysisTool {
+
+    constructor(private toolName: string, protected buildOutput: BuildOutput, private uiInputName: string) {
+
+    }
+
+    /**
+     * This method lets implementers specify where the reports are located
+     * 
+     * @protected
+     * @abstract
+     * @param {ModuleOutput} output
+     */
+    protected abstract getBuildReportDir(output: ModuleOutput);
+
+    /**
+     * Report parser that extracts the number of affected files and the number of violations from a report
+     * 
+     * @protected
+     * @abstract
+     * @param {string} xmlReport
+     * @param {string} moduleName
+     * @returns {[number, number]} a tuple of [affected_file_count, violation_count]
+     */
+    protected abstract parseXmlReport(xmlReport: string, moduleName: string): [number, number];
+
+    public configureBuild(toolRunner: ToolRunner): ToolRunner {
+        switch (this.buildOutput.buildEngine) {
+            case BuildEngine.Maven: {
+                throw new Error('Not implemented');
```

Comment:
Are these Errors caught? If not, they represent user-visible strings and should be localised.